### PR TITLE
build: switch to ruff linter

### PR DESCRIPTION
**Pull request by patrit**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -40,3 +38,60 @@ build-backend = "poetry.core.masonry.api"
...
+"C901", "PLR0913", # ignore complexity
+"D",
+"COM812", "ISC001",
+"EM101", "EM102", 
+"UP007", # Optional[T] is ok atm. later: move to T | None 
+ "S101", "TD", "TRY003"]
+
+[tool.ruff.per-file-ignores]
+"**/__init__.py" = ["F401"]
+# the folowing exclusionn have beenn introduced to prevent huge changes
```
Comment (patrit) [inline pyproject.toml:58]: slightly nasty, but easy to make the transition and keep the patch "small"

**Review by christiansiegel — APPROVED** [review]

**Comment by patrit** [comment]
We may keep mypy .c.f https://docs.astral.sh/ruff/faq/#how-does-ruff-compare-to-mypy-or-pyright-or-pyre

File: pyproject.toml (hunk 1 above)
Comment (patrit) [reply pyproject.toml:58]: typos.

**Comment by christiansiegel** [comment]
> We may keep mypy .c.f https://docs.astral.sh/ruff/faq/#how-does-ruff-compare-to-mypy-or-pyright-or-pyre

ooh, I was under the impression that it can do both. Please keep mypy then

**Review by christiansiegel — APPROVED** [review]
lgtm 🚀

**Pull request merged**